core/templates/dev/head/pages/exploration_player/PlaythroughService.js

```diff
@@ -198,29 +199,29 @@ oppia.factory('PlaythroughService', [
     };
 
     return {
-      initSession: function(newExplorationId, newExplorationVersion) {
-        isPlayerInSamplePopulation = _determineIfPlayerIsInSamplePopulation();
+      initSession: function(
+          explorationId, explorationVersion, playthroughProbability,
+          whitelistedExpIds) {
+        isPlayerInSamplePopulation = _determineIfPlayerIsInSamplePopulation(
+          playthroughProbability);
+        explorationWhitelisted = this.isExplorationWhitelisted(
+          explorationId, whitelistedExpIds);
         playthrough = PlaythroughObjectFactory.createNew(
-          null, newExplorationId, newExplorationVersion, null, {}, []);
+          null, explorationId, explorationVersion, null, {}, []);
         expStopwatch = StopwatchObjectFactory.create();
       },
       isPlayerExcludedFromSamplePopulation: function() {
         return !isPlayerInSamplePopulation;
       },
-      isExplorationWhitelisted: function(expId) {
-        var whiteListedExpIds =
-          constants.WHITELISTED_EXPLORATION_IDS_FOR_SAVING_PLAYTHROUGHS;
-        if (whiteListedExpIds.indexOf(expId) !== -1) {
-          return true;
-        }
-        return false;
+      isExplorationWhitelisted: function(expId, whitelistedExpIds) {
```

Comment:
I think storing the `whitelistedExpIds` makes more sense than storing the return value of `isExplorationWhitelisted`.

Reply:
Done,